content/MANUAL.md

```diff
@@ -1,61 +1,316 @@
+- `chore: update dependencies`
+
+Note: a commit that doesn't follow the above guidelines will throw an error
+
+## Building and Testing
+
+### Running the Demo
+
+The demo application helps you test your binding in a real-world scenario:
 
 ```bash
+# On Unix
 ./build.sh demo --watch
+
+# On Windows
+build.cmd demo --watch
+```
+
+This will:
+1. Build your binding project
+2. Start a development server with hot reloading
+3. Automatically open a browser with your demo application
+
+The development server is configured to:
+- Provide hot module replacement (changes appear instantly)
+- Automatically open the browser when started
+- Ignore F# source files in watch mode (to prevent double compilation)
+
+### Build Commands
+
+The build script supports various commands:
+
+```bash
+# Get help
+./build.sh --help    # Unix
+build.cmd --help     # Windows
+
+# List all available commands
+./build.sh list      # Unix
+build.cmd list       # Windows
+
+# Run tests (if you've added them)
+./build.sh test      # Unix
+build.cmd test       # Windows
 ```
 
-## Make a release
+## Deployment and Publishing
 
-You need to have a NuGet API saved in `NUGET_KEY` environment variable, you can get it from https://www.nuget.org/account/apikeys.
+### Publishing to NuGet
 
-To make a release, you can use the following command:
+Before publishing, you need a NuGet API key saved in the `NUGET_KEY` environment variable. Get one from https://www.nuget.org/account/apikeys.
 
 ```bash
-./build.sh release
+# Create a release with default settings
+./build.sh release      # Unix
+build.cmd release       # Windows
+
+# Create a specific version release
+./build.sh release --version 1.2.3    # Unix
+build.cmd release --version 1.2.3     # Windows
+
+# Show all release options
+./build.sh release --help    # Unix
+build.cmd release --help     # Windows
+```
+
+The release process:
+1. Validates your project
+2. Builds the package
+3. Publishes to NuGet
+4. Creates a GitHub release (if configured)
+5. Deploys the demo to GitHub Pages (if enabled in Directory.UserConfig.props)
+
+### GitHub Pages Deployment
+
+GitHub Pages allows you to host a static website directly from your GitHub repository. This is ideal for showcasing your binding with an interactive demo.
+
+#### What is GitHub Pages?
 
-# Check the help for more information
-./build.sh release --help
+GitHub Pages is a free hosting service that takes HTML, CSS, and JavaScript files straight from a repository on GitHub, optionally runs the files through a build process, and publishes a website.
+
+#### Setting Up GitHub Pages for Your Repository
+
+1. **Enable GitHub Pages in your repository**:
+   - Go to your repository on GitHub
+   - Navigate to Settings > Pages
+   - Set the Source to "GitHub Actions" (recommended) or "Deploy from a branch"
+   
+2. **Configure deployment in your project**:
+   - In `Directory.UserConfig.props`, ensure `<GitHubDeployment>true</GitHubDeployment>` is set
```

Comment:
This is the first time I learn about this `GitHubDeployment` deployment should work without it in theory because this is done via CLI directly.